vircadia/vircadia
# Fix HIFI_MEMORY_DEBUGGING on Linux/gcc

It seems that libasan was running into some sort of trouble due to static linking.

Additionally, `-fstack-protector-strong` has been enabled by disabling the usage of the `FIR_1x4_AVX512` function under memory debugging. This should provide additional protection against stack smashing.

Testing:
* `export VIRCADIA_MEMORY_DEBUGGING=1`
* build on Linux. OSX may also work, but the build probably takes the clang path there, which wasn't changed in this PR. Windows isn't supported yet.
* Verify that the interface starts (before it wouldn't). 
* With this enabled, you should see additional errors in the log from the additional sanitizer checks, such as:
`vircadia-master/libraries/entities-renderer/src/RenderableGridEntityItem.cpp:53:9: runtime error: load of value 190, which is not a valid value for type 'bool'`
* The interface crashing with an error such as this is an acceptable outcome. It means the sanitizer is working and found something to fix:
`==306449==ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7ffc102c4640 at pc 0x7f7957533f19 bp 0x7f78651ba030 sp 0x7f78651ba028`
* Noticeably lower performance is a very expected side-effect of doing additional checking.

Fixes #1119
